pkgs/racket-doc/scribblings/reference/threads.scrbl

```diff
@@ -195,10 +212,19 @@ otherwise.}
 @;------------------------------------------------------------------------
 @section[#:tag "threadsync"]{Synchronizing Thread State}
 
-@defproc[(thread-wait [thd thread?]) void?]{
+@defproc[(thread-wait [thd thread?]
+                      [fail-k (procedure-arity-includes/c 0) void])
+         any]{
 
 Blocks execution of the current thread until @racket[thd] has
-terminated. Note that @racket[(thread-wait (current-thread))]
+terminated. If the thread's procedure raised an exception or otherwise
+aborted to the thread's initial @tech{prompt}, @racket[fail-k] is
+called to produce the result of @racket[thread-wait]. Otherwise, if
+the thread records its results (see @racket[#:keep] in
+@racket[thread]), those results are returned, while @|void-const| is
+return if the thread does not keep its results.
```

Comment:
This program prints `#<void>`. Does this agree with the above description?

```
#lang racket

(define t
  (thread
   #:keep 'results
   (λ ()
     (sleep 5)
     1)))

(break-thread t)
(println (thread-wait t (λ () 123)))
```

Reply:
That example was misbehaving due to s bug, now fixed.